### PR TITLE
Apply prettier to all non-docs files

### PR DESCRIPTION
**Pull request by jerelmiller**
Run prettier formatting on all non-docs files to get the full codebase using automatic formatting.

### Checklist:

- [x] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [x] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Comment by jerelmiller** [comment]
I spot checked all files and added `prettier-ignore` comments where necessary (yes, all 201 files).

**IMPORTANT** This PR should be _merged_ not squashed in order to preserver the commit hash in `git-blame-ignore-revs` without a commit to `main`.

File: src/__tests__/__snapshots__/exports.ts.snap [hunk 1]
```diff
@@ -211,7 +211,6 @@ Array [
 
 exports[`exports of public entry points @apollo/client/link/persisted-queries 1`] = `
 Array [
-  "PersistedQueryLink",
```
Comment (phryneas) [inline src/__tests__/__snapshots__/exports.ts.snap:214]: This is curious - how did we lose an export here?

File: src/react/hooks/useBackgroundQuery.ts [hunk 2]
```diff
@@ -18,21 +18,21 @@ import type { SkipToken } from "./constants.js";
 
 export type UseBackgroundQueryResult<
   TData = unknown,
-  TVariables extends OperationVariables = OperationVariables
+  TVariables extends OperationVariables = OperationVariables,
```
Comment (phryneas) [inline src/react/hooks/useBackgroundQuery.ts:21]: How come we have changes in these files? They should have been formatted nicely before.

Comment (phryneas) [reply src/react/hooks/useBackgroundQuery.ts:21]: Huh, seems the `.prettierignore` we had in place in-between was missing a `!/src` 🤦

File: src/__tests__/__snapshots__/exports.ts.snap (hunk 1 above)
Comment (phryneas) [reply src/__tests__/__snapshots__/exports.ts.snap:214]: Okay, this is interesting - the `;` makes quite the difference here:

image

vs

image

**Review by phryneas — APPROVED** [review]
I love it!

With the mystery around `PersistedQueryLink` solved, I'd say let's get this merged ASAP before we get a load of merge conflicts :)

Disclaimer: I ignored `__tests__` files in my review, and mostly skimmed the rest. This is quite a lot.

File: src/__tests__/__snapshots__/exports.ts.snap (hunk 1 above)
Comment (jerelmiller) [reply src/__tests__/__snapshots__/exports.ts.snap:214]: Good find! Thanks for digging into this more :)

**Pull request merged**